tests/framework/icd/test_icd.cpp

```diff
@@ -427,9 +432,25 @@ PFN_vkVoidFunction get_instance_func(VkInstance instance, const char* pName) {
 }
 
 PFN_vkVoidFunction get_device_func(VkDevice device, const char* pName) {
+    bool found = false;
+    PhysicalDevice* found_phys_dev{};
+    for (auto& phys_dev : icd.physical_devices) {
+        for (auto& device_handle : phys_dev.device_handles) {
+            if (device_handle == device) {
+                found = true;
+                found_phys_dev = &phys_dev;
+                break;
+            }
+        }
+    }
+    if (!found) return nullptr;
     if (string_eq(pName, "vkDestroyDevice")) return TO_VOID_PFN(test_vkDestroyDevice);
     if (string_eq(pName, "vkCreateSwapchainKHR")) return TO_VOID_PFN(test_vkCreateSwapchainKHR);
-
+    for (auto& function_name : found_phys_dev->known_device_functions_no_implementation) {
+        if (string_eq(pName, function_name)) {
+            return reinterpret_cast<PFN_vkVoidFunction>(test_stub_func_with_return);
```

Comment:
NOTE: Some vulkan functions do not have a return, so we might want to have some kind of mechanism long-term to handle that.

Reply:
Hmm there is a 'no-return-stub-function' but I just threw them all together since they aren't suppose to do anything. I'm not sure that the distinction is warrented because I think it's UB to pass arguments to a function pointer which takes 0 arguments.